projects/components/src/stepper/stepper.component.ts

```diff
@@ -34,17 +35,20 @@ import { StepperTabComponent } from './tab/stepper-tab.component';
           <-- NOTE: completed and step control both should not be present when in linear flow. So when step control is
           provided, we explicitly set completed to false so that the stepper can work with teh form validity status -->
           <mat-step
-            [completed]="step?.stepControl ? false : step.completed"
+            [completed]="step?.stepControl?.valid ?? step.completed"
             [stepControl]="step?.stepControl"
             [optional]="step.optional"
             [editable]="step.editable"
           >
             <ng-template matStepLabel>
-              <ht-label class="header-label" [label]="step.label"></ht-label>
+              <ht-label class="header-label" [attr.data-editable]="step.editable" [label]="step.label"></ht-label>
```

Comment:
why do we have to use attributes for this? Can we do it by setting a css class?

Reply:
We can.